src/Http/Http/src/Features/RequestServicesFeature.cs

```diff
@@ -41,11 +42,25 @@ public IServiceProvider RequestServices
             }
         }
 
-        public void Dispose()
+        public async ValueTask DisposeAsync()
         {
-            _scope?.Dispose();
+            switch (_scope)
+            {
+                case IAsyncDisposable asyncDisposable:
+                    await asyncDisposable.DisposeAsync();
+                    break;
+                case IDisposable disposable:
+                    disposable.Dispose();
+                    break;
+            }
+
             _scope = null;
             _requestServices = null;
         }
+
+        public void Dispose()
+        {
+            DisposeAsync().ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
So you want this to throw?

Reply:
I'm ok with this not throwing. It should be super rare now since we by default will dispose it async.